riscv/mmu.cc

```diff
@@ -139,12 +149,37 @@ bool mmu_t::mmio_store(reg_t addr, size_t len, const uint8_t* bytes)
   return sim->mmio_store(addr, len, bytes);
 }
 
-void mmu_t::load_slow_path(reg_t addr, reg_t len, uint8_t* bytes, uint32_t xlate_flags)
+void mmu_t::check_triggers(triggers::operation_t operation, reg_t address, bool has_data, reg_t data)
```

Comment:
Instead of `has_data/data` how about `std::optional<reg_t> data`? Then the code can never accidentally use `data` when it's meaningless.

Reply:
Agreed… I would’ve written it that way in Scala.

Reply:
It's a larger refactoring than I thought, so I'll make it a second PR.